hazelcast/src/test/java/com/hazelcast/internal/partition/PartitionRebalanceTest.java

```diff
@@ -0,0 +1,187 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.partition;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.map.IMap;
+import com.hazelcast.spi.properties.ClusterProperty;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.hazelcast.internal.partition.TestPartitionUtils.assertAllPartitionsBelongTo;
+import static com.hazelcast.internal.partition.TestPartitionUtils.assertSomePartitionsBelongTo;
+import static com.hazelcast.internal.partition.TestPartitionUtils.dumpPartitionTable;
+import static com.hazelcast.test.Accessors.getPartitionService;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class PartitionRebalanceTest extends HazelcastTestSupport {
+
+    // maximum time to wait for assertions after partition rebalancing
+    // if tests fail on busy build machines, this may have to be increased
+    private static final int PARTITION_REBALANCE_TIMEOUT = 5;
+
+    @Rule
+    public final Timeout timeout = Timeout.seconds(20);
+
+    private HazelcastInstance instance1;
+    private HazelcastInstance instance2;
+    TestHazelcastInstanceFactory factory;
+
+    @Before
+    public void setup() {
+        factory = createHazelcastInstanceFactory();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void configureWithNegativeDelay() {
+        instance1 = factory.newHazelcastInstance(getConfig(-3));
+    }
+
+    @Test
+    public void rebalanceOccursAfterDelay_whenDelayed() {
+        // this test will take > 10 seconds :(
+        instance1 = factory.newHazelcastInstance(getConfig(PARTITION_REBALANCE_TIMEOUT));
+        getPartitionService(instance1).firstArrangement();
+
+        AtomicBoolean done = new AtomicBoolean(true);
```

Comment:
I was wondering what purpose `done` flag plays currently; there is `spawnPartitionTableDumper` which accepts `AtomicBoolean done` as a parameter, but I think that's never called

Reply:
good catch, seems like a leftover from testing